src/AsyncUtils.js

```diff
@@ -30,20 +30,26 @@ export async function* filter<T>(
 
 export class AsyncQueue {
   values: any[];
+  cleanedUp: boolean = false;
   promise: Promise<void>;
   unsubscribe: () => void;
   resolvePromise: () => void;
   iterable: AsyncGenerator<any, void, void>;
 
   constructor(unsubscribe: () => void) {
-    this.unsubscribe = unsubscribe;
-
     this.values = [];
+    this.unsubscribe = unsubscribe;
     this.createPromise();
 
     this.iterable = this.createIterable();
   }
 
+  close(): void | Promise<void> {
```

Comment:
why `| Promise<void>`?

Reply:
I tried to make the subscriber type a bit looser, e.g. not need to be async. This is was loosened to allow passing the queues through sort of interchangeably

Reply:
oh but this isn't the subscriber right? and this isn't implementing an interface type